test/cmdlineargs.jl

```diff
@@ -277,21 +270,33 @@ let exename = `$(Base.julia_cmd()) --precompiled=yes`
     end
 end
 
+let exename = `$(Base.julia_cmd()) --precompiled=yes`
+    # --startup-file
+    # Note: cannot cleanly test for --startup-file=yes, because errors or deprication
+    # warnings in ~/.juliarc.jl cause the tests to freeze
+    let JL_OPTIONS_STARTUPFILE_ON = 1,
+        JL_OPTIONS_STARTUPFILE_OFF = 2
+        # @test parse(Int,readchomp(`$exename -E "Base.JLOptions().startupfile" --startup-file=yes`)) == JL_OPTIONS_STARTUPFILE_ON
```

Comment:
This is a bit of a problem.  I don't know how to properly test that the startup file is being used, if a broken startup file causes the tests to hang.